dallinger/recruiters.py

```diff
@@ -75,6 +78,38 @@ def run_status_check():
         session.commit()
 
 
+@dataclass
+class RecruitmentStatus:
+    """
+    Class for standardized status reporting of recruitments of various recruiters.
+
+    Args:
+        recruiter_name (str): The name of the recruiter
+        participant_status_counts (dict): A histogram in dictionary format where the keys are the statuses of submissions
+        (e.g., `"APPROVED"` or `"REJECTED"`) and the values are the respective counts
+        study_id (str): The ID used on the recruiting platform
+        study_status (str): Status of the recruitment, e.g.,  `"ACTIVE"` or `"AWAITING REVIEW"` are valid study statuses on Prolific
+        study_cost (float): Total cost for a recruitment that includes both base payments (rewards on Prolific) and bonuses as well as service fees and taxes if returned by the API
+        metadata (dict): Dictionary of any information specific to the recruiter, e.g. for Prolific the median duration of approved participants and the wage_per_hour computed by the platform
+    """
+
+    recruiter_name: str
+    participant_status_counts: dict
+    study_id: str
+    study_status: str
+    study_cost: float
+    metadata: dict
+
+    def to_dict(self):
+        data = self.__dict__
+        recruiter_name = data.pop("recruiter_name")
```

Comment:
Seems dangerous to call the destructive operation `pop` on `self.__dict__`. I'd probably write instead

```py
def to_dict(self):
    result = {
        "recruitment_" + k: v for k, v in self.__dict__
         if k != "recruiter_name"
    }
    result["recruiter"] = self.recruiter_name
    return result
```

That said, I'm not 100% about doing this `recruiter_` logic inside the class definition. If it were me, I would just leave the dataclass with its builtin `asdict` method, and then do any formatting like this in the place where the `RecruitmentStatus` object is being used (since it is context-specific). Doesn't really matter though.